### ISSUE TITLE
Add BrowserWindow 'show' and 'hide' events

### ISSUE THREAD

**Issue by feross** [atom/electron#4693]
Would like to do:

```js
var win = new electron.BrowserWindow()
win.on('show', function () {})
win.on('hide', function () {})
```

This PR (https://github.com/atom/electron/pull/862) added lots of events, but not `'show'` or `'hide'`.

**Comment by zeke** [comment]
Related: https://github.com/atom/electron/pull/3697

**Comment by feross** [comment]
@zeke Isn't that issue about `app.show()`/`app.hide()`? I think that's slightly different than `win.show()`/`win.hide()`, right?

**Comment by phamdaniel** [comment]
Closed with #4700
